scenarios/kubernetes_verify.py

```diff
@@ -98,12 +120,50 @@ def main(branch, script, force, on_prow):
             'bash', '-c', 'cd kubernetes && %s' % script,
         )
     else:
-        os.environ["REPO_DIR"] = k8s # hack/lib/swagger.sh depends on this
-        os.environ["KUBE_FORCE_VERIFY_CHECKS"] = str(force)
-        os.environ["KUBE_VERIFY_GIT_BRANCH"] = str(branch)
-        check(
-            'bash', '-c', 'cd kubernetes && %s' % script,
-        )
+        # make sure we are cd-ed to the repo
+        k8s = '/go/src/k8s.io/kubernetes'
+        os.chdir(k8s)
+
+        # setup env
+        os.environ['REPO_DIR'] = k8s # hack/lib/swagger.sh depends on this
+        os.environ['KUBE_FORCE_VERIFY_CHECKS'] = str(force)
+        os.environ['KUBE_VERIFY_GIT_BRANCH'] = str(branch)
+
+        # TODO(bentheelder): mimic the other script too, and eventually
+        # migrate this flag to be a bool between the two possible scripts instead.
+        if script == './hack/jenkins/verify-dockerized.sh':
+            # mimic k8s.io/kubernetes/hack/jenkins/verify-dockerized.sh
+            # but patched slightly to work appropriately on prow
+            path_prefix = os.environ['GOPATH'] + '/bin'
+            path_prefix += ':' + os.getcwd() + '/third_party/etcd:/usr/local/go/bin'
+            os.environ['PATH'] = path_prefix + ':' + os.environ['PATH']
+            os.environ['ARTIFACTS_DIR'] = os.environ['WORKSPACE'] + '/artifacts'
+            os.environ['LOG_LEVEL'] = '4'
+
+            # TODO(BenTheElder): remove this horrible debug hack
+            # patch build/common to properly not rsync in .git
+            check('bash', '-c',
+                  'sed -i \'s#/.git/#/.git#\' ./build/common.sh && cat ./build/common.sh')
+            # make a dummy git commit so that the working tree is "clean"
+            # as far as the verify scripts are concerned
+            check('bash', '-c', 'git add . && git commit -m "dummy commit"')
```

Comment:
My biggest issue with this is that CI needs to create a dummy commit (outside of a container).

I think the right fix is to put this into k/k. Which branches would we be looking to move over to this prow verify job?

Reply:
I force pushed with a comment fix and GitHub ate my previous comment, I don't want to check this in as is:
https://github.com/kubernetes/test-infra/pull/5466#discussion_r150379305
https://github.com/kubernetes/test-infra/pull/5466#issuecomment-343649320

I agree, probably the best thing is to put this into k/k. The goal is to move everything to Prow so at least 1.7, 1.8, master, probably 1.6 too.